2016/proposals/result-aggregation-server-garg-prerit.md

```diff
@@ -0,0 +1,196 @@
+ 
+## Technical Details 
+I am breaking it down to several steps in which this project has to be approached : 
+### System Information collection 
+The installation-test script need to be modified to retrieve some useful information about the user and the environment in which he is running the installation-test-script. 
+* Operating System information, which includes : 
+  * System information ( Linux / windows / Mac etc ) 
+  * Distribution information ( redhat / debian / fedora etc) 
+  * Machine Architecture information ( x86 / x64 ) 
+  * Complete platform information 
+  * Uname ( System info like kernel-name, kernel-version etc ) 
+  * Version ( System’s release version ) 
+  * Machine information ( Machine type, e.g. 'i386' ) 
+  
+All of this information can be retrieved by using a library in python called "Platform", code of which I wrote in my [demo](https://github.com/prerit2010/DemoAPI) application. The code snippet : 
+``` 
+import platform 
+ 
+system_dist = str(platform.dist()) 
+system = platform.system() 
+machine = platform.machine() 
+system_platform = platform.platform() 
+uname = platform.uname() 
+version = platform.version() 
+``` 
+* Compilation environment used by user 
+ * Python Version 
+ * Python Build 
+ * Python Compiler 
+ * Python implementation ( e.g. ‘CPython’, ‘IronPython’, ‘Jython’, ‘PyPy’ ) 
+  
+All of which can be retrieved by the same library "platform" : 
+``` 
+import platform 
+ 
+python_version = platform.python_version() 
+python_build = platform.python_build() 
+python_compiler = platform.python_compiler() 
+python_implem = platform.python_implementation() 
+``` 
+Sample of output generated in case of Ubuntu : 
+``` 
+Python version: ['2.7.3 (default, Sep 26 2012, 21:51:14) ', '[GCC 4.7.2]'] 
+dist: ('Ubuntu', '12.10', 'quantal') 
+linux_distribution: ('Ubuntu', '12.10', 'quantal') 
+system: Linux 
+machine: x86_64 
+platform: Linux-3.5.0-22-generic-x86_64-with-Ubuntu-12.10-quantal 
+uname: ('Linux', 'stijn-W150HRM', '3.5.0-22-generic', '#34-Ubuntu SMP Tue Jan 8 21:47:00 UTC 2013', 'x86_64', 'x86_64') 
+version: #34-Ubuntu SMP Tue Jan 8 21:47:00 UTC 2013 
+``` 
+Some or most of these information are being retrieved in the [installation-test-scripts](https://github.com/wking/swc-setup-installation-test). I would just need to collect them. 
+### Success/Failure of packages information collection 
+* **Successful Installs** list is already provided by the script. A list ```successes``` contains both checker and the version. All I need to do is grab the full name by ```checker.full_name()``` and ```version``` of the successfully installed packages. 
```

Comment:
Are *you* grabbing the data, or is this the server's role?

Also: does triple "\`" compile for you inline in Markdown? I think we inline code mark is a pair of ticks "`" with code between them (but I may be wrong).